### ISSUE TITLE
Introduce AWS provider for airy create

### ISSUE THREAD

**Issue by ljupcovangelski** [airyhq/airy#1018]
## Open questions

- [x] Do we use eks? 
> Yes
- [x] What are the required credentials and how do we require them? (env seems the most common approach there)
> There will be a `aws-profile` flag on the command
- [x] What else do we need to actually make this provider work? (For example: load balancer?)
> If a VPC and 2 subnets are already created, then we need an EKS cluster, 1 Node and 1 Load Balancer

**Comment by lucapette** [comment]
this depends on #1017 (I know we have a branch about this... but it's really not a good plan to do this first)

**Comment by paulodiniz** [comment]
About the topics that @ljupcovangelski mentioned

> We use eks?

I think so. Or better put: I don't see a reason not to

> What are the required credentials and how do we require them? (env seems the most common approach there)

I think having a valid `~/.aws/config` could be a requirement for this provider. What do you think?

> What else do we need to actually make this provider work? (For example: load balancer?)

From my AWS understanding

1. Create a VPC
2. Create two subnets in different regions (`us-east-1a` and `us-east-1b` for example). Both subnets must be contained in the VPC
3. Create a Arn Role
4. Create the cluster
5. Create a Load Balancer

**Comment by ljupcovangelski** [comment]
I agree with @paulodiniz that those are the steps needed to make this provider work. I just want to add two notes:
- Apart from creating the EKS cluster, we also need to add nodes/instances to the cluster. How many instances and which size would we recommend? Would be good to make this configurable and in correlation with some resources request/limits of the components.
- Perhaps we can leave an option for people to use an existing VPC, which will facilitate connecting of the EKS cluster (and the components) to the prerequisites, in case somebody is running Kafka/PostgreSQL and Redis on their own and in an already running cluster.

For the other two points:
- I think we should use an EKS
- For the profile, I think we can introduce a flag `--aws-profile`, connected also to an ENV variable `AWS_PROFILE`

**Comment by paulodiniz** [comment]
@ljupcovangelski Good point about adding the nodes!

On adding the `vpc` options, I think it's a nice to have, but we could go without it.

I don't understand why the `AWS_PROFILE`, isn't it easier to rely on the aws config file itself?

**Comment by ljupcovangelski** [comment]
> I don't understand why the `AWS_PROFILE`, isn't it easier to rely on the aws config file itself?

Because in the config file, one can have multiple profiles configured.

**Comment by paulodiniz** [comment]
@ljupcovangelski That's true 👍 Thanks for the clarification

**Comment by lucapette** [comment]
> I agree with @paulodiniz that those are the steps needed to make this provider work. I just want to add two notes:
> 
>     * Apart from creating the EKS cluster, we also need to add nodes/instances to the cluster. How many instances and which size would we recommend? Would be good to make this configurable and in correlation with some resources request/limits of the components.

I'd default to one and then have follow up issues we do in a "AWS provider milestone" (chances are we'll have a bunch of follow up issues for things that are not critical to the airy create command "beta version" so to speak)

> 
>     * Perhaps we can leave an option for people to use an existing VPC, which will facilitate connecting of the EKS cluster (and the components) to the prerequisites, in case somebody is running Kafka/PostgreSQL and Redis on their own and in an already running cluster.

I think I'd build this first actually (like we're doing with the general command and the kube.conf. Build a working version with a "bad UI" first and when everything is wired make it smarter). Let them pass a vpc and then we make it smarter in "provider milestones"

> 
> 
> For the other two points:
> 
>     * I think we should use an EKS
> 
>     * For the profile, I think we can introduce a flag `--aws-profile`, connected also to an ENV variable `AWS_PROFILE`

👍

**Comment by paulodiniz** [comment]
If we decide to use a flag for an existing VPC, we must also provide two additional ones for the two required subnets (in different regions) that AWS requires

Maybe creating both VPC and subnets is the "easiest" way to go now. What do you think?

**Comment by lucapette** [comment]
done in #1240
